desktop-widgets/importgps.cpp

```diff
@@ -0,0 +1,329 @@
+// SPDX-License-Identifier: GPL-2.0
+#include "ui_importgps.h"
+#include "desktop-widgets/importgps.h"
+#include "desktop-widgets/locationinformation.h"
+#include "core/dive.h"
+
+#include <time.h>
+#include <QFileDialog>
+#include <QProcess>
+#include <QDateTime>
+#include <QDate>
+#include <stdlib.h>
+#include <stdio.h>
+#include <QDebug>
+
+#define BUFFER_SIZE 5000
+
+/* Import dive coordinates from a GPS device and synchronise them with the dive profile information
+   of a dive computer. This file contains the infrastructure to:
+   1) Read a .GPX file from a GPS system.
+   2) Find the first gpx trackpoint that follows after the start of the dive.
+   3) Allow modification of the coordinates dependent on international time zone and
+              on differences in local time settings between the GPS and the dive computer.
+   4) Saving the coordinates into the Coordinates text box in the Dive Site Edit panel
+              and which which causes the map to show the location of the dive site. 
+   The structure coords is used to store critical information.  */
+ImportGPS::ImportGPS(QWidget *parent, QString fileName, class Ui::LocationInformation *LocationUI) : QDialog(parent),
+	fileName(fileName), LocationUI(LocationUI)
+{
+	ui.setupUi(this);
+	connect(ui.timeDiffEdit, SIGNAL(timeChanged(const QTime)), this, SLOT(timeDiffEditChanged()));
```

Comment:
I would so much prefer the pointer-to-member version, because it gives compile-time type checking. Though we can do that later.

Reply:
Please give a bit more detailed information. I would like to correct this now.

Reply:
there are two different ways to do connect calls. the one you have here with the string based syntax, and the "modern" one which uses type checking at compile time. The difference is well explained in the Qt Wiki: https://wiki.qt.io/New_Signal_Slot_Syntax